### DISCUSSION TITLE
Regression of `brew doctor` reporting `no formulae` warning on third party formula conflicting name between taps

### DISCUSSION THREAD

**Discussion by itchyny** [Homebrew/discussions#5156] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.9-114-ge9cb65b
ORIGIN: https://github.com/Homebrew/brew
HEAD: e9cb65bfbaf4e0d531e598f960d8e738d9f66584
Last commit: 6 hours ago
Core tap HEAD: b61c5ff0561e1450fdbad0990daf0391d355a717
Core tap last commit: 42 minutes ago
Core tap JSON: 25 Feb 02:56 UTC
Core cask tap HEAD: 93229f04fe866d8a64ea939b2ed9094f99627485
Core cask tap last commit: 73 minutes ago
Core cask tap JSON: 25 Feb 02:56 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.a24GjbGsaP/org.xquartz:0
HOMEBREW_EDITOR: vim
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_NO_ENV_HINTS: set
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.44.0 => /opt/homebrew/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.2.1-arm64
CLT: 15.1.0.0.1.1700200546
Xcode: 15.2
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed kegs have no formulae!
This means they were either deleted or installed manually.
You should find replacements for the following formulae:
  mmv
```

### Description of issue

Previously, the `brew doctor` did not warn about the `mmv` formula, installed from `itchyny/tap/mmv`. However, recent commit, specifically https://github.com/Homebrew/brew/commit/d3ef383753324c51c9218d33a552c621b96a8c7e, changed the behavior of `brew doctor`. Brew starts to report warning on the formula. I know there is an unfortunate name conflict against `homebrew-core`, but that's different issue, and since the formula was installed by `itchyny/tap/mmv`, `brew doctor` should not emit this warning. I expect the `brew doctor` command reports that it is clean. I can't check the issue template verifications, I open a discussion.

Reproduction commands:
```
brew uninstall mmv # just in case
brew install itchyny/tap/mmv
brew doctor
```

**Comment by aramprice** [comment]
I have been seeing the same warning for [rcm](https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/r/rcm.rb) recently.

**Comment by itchyny** [comment]
This issue has been resolved by https://github.com/Homebrew/brew/pull/16879, so closing.
